src/gui/executionlogwidget.h

```diff
@@ -30,33 +30,39 @@
 #define EXECUTIONLOGWIDGET_H
 
 #include <QWidget>
-#include "base/logger.h"
 
-class LogListWidget;
+#include "base/logger.h"
 
 namespace Ui
 {
     class ExecutionLogWidget;
 }
 
+class BaseLogModel;
+class LogFilterModel;
+class LogListView;
+class LogMessageModel;
+class LogPeerModel;
+
 class ExecutionLogWidget : public QWidget
 {
     Q_OBJECT
 
 public:
-    ExecutionLogWidget(QWidget *parent, const Log::MsgTypes &types);
-    void showMsgTypes(const Log::MsgTypes &types);
+    ExecutionLogWidget(QWidget *parent, Log::MsgTypes types);
     ~ExecutionLogWidget();
-
-private slots:
-    void addLogMessage(const Log::Msg &msg);
-    void addPeerMessage(const Log::Peer &peer);
+    
+    void setMsgTypes(Log::MsgTypes types);
 
 private:
-    Ui::ExecutionLogWidget *m_ui;
+    void displayContextMenu(const QPoint &pos, const LogListView *view, const BaseLogModel *model) const;
 
-    LogListWidget *m_msgList;
-    LogListWidget *m_peerList;
+    Ui::ExecutionLogWidget *m_ui;
+    LogMessageModel *m_messageModel;
+    LogPeerModel *m_peerModel;
+    LogFilterModel *m_messageFilterModel;
+    LogListView *m_messageView;
```

Comment:
Seems you don't need fields for views.

Reply:
If I understand it correctly by that logic I don't need to store LogMessageModel and LogPeerModel as well, so I removed them as well